autotest/ogr/ogr_geojson.py

```diff
@@ -3525,6 +3525,58 @@ def test_ogr_geojson_non_finite():
     gdal.Unlink(tmpfilename)
 
 
+###############################################################################
+# Test writing fields with and without automatic JSON interpretation
+
+
+def test_ogr_geojson_json_string_autodetect():
+
+    json_content = """{
+  "type": "FeatureCollection",
+  "features": [
+      { "type": "Feature", "properties": { "jsonish": "[nan]" }, "geometry": null }
+  ]
+}"""
+    with gdal.quiet_errors():
+        ds = ogr.Open(json_content)
+    if ds is None:
+        # Might fail with older libjson-c versions
+        pytest.skip()
```

Comment:
wouldn't using ``[5]`` instead of ``[nan]`` avoid issues with older libjson-c ?
```suggestion
      { "type": "Feature", "properties": { "jsonish": "[5]" }, "geometry": null }
  ]
}"""
    ds = ogr.Open(json_content)
```

Reply:
Thanks. I'll make that change. I only used `[nan]` because that was the string value that had brought the automatic conversion of strings to JSON to my attention.

Reply:
what about just using ``assert f["jsonish"] == "[5]"`` to make things more compact & pytest-idiomatic ? (we don't really need to output the full feature upon test failure)

Reply:
Sure, I'll be happy to make that change. I only did any of the things that this test does because I was copying and pasting from the test above.

Reply:
> Sure, I'll be happy to make that change. I only did any of the things that this test does because I was copying and pasting from the test above.

sure. We gradually try to improve things, at least in new code. Long process... :-)

Reply:
Thanks for the style tips! I wish I knew the GDAL test conventions better.